### DISCUSSION TITLE
Asciinema + text search

### DISCUSSION THREAD

**Discussion by amityahav** [asciinema/discussions#249] [Show and tell]
Hey @ku1ik im happy to share that we are using asciinema in production in order to audit ssh sessions in our platform.
one of the requests from our customers was to be able to search upon those recordings and after couple of months i present the result here in the video: 

https://github.com/asciinema/discussions/assets/74831037/584a6fa9-9ecd-449b-9a4d-58725e171208

i've modified abit the normal .cast file in order to be 
able to capture the user's input based from the output events
then when the session ends the .cast file is processed with a terminal emulator that outputs the prompts and index them in our indexing engine. it was a fun task and we are happy with the result

**Comment by amityahav** [comment]
@ku1ik  in case you've missed it :)

**Comment by ku1ik** [comment]
This is cool! At first I thought this is a purely server-side search ... but looks like you modified the player as well.

So it searches only the prompt lines, right?

**Reply by amityahav** [reply]
yes, we add annotations to the cast events to properly catch prompts with as little noise as possible. than we feed it again to a terminal emulator, build the grid and catch the output of the terminal that contains the prompt.
